Applications/DataExplorer/DataView/ProcessModel.cpp

```diff
@@ -0,0 +1,179 @@
+/**
+ * \file
+ * \author Karsten Rink
+ * \date   2010-10-18
+ * \brief  Implementation of the ProcessModel class.
+ *
+ * \copyright
+ * Copyright (c) 2013, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "ProcessModel.h"
+
+// ThirdParty/logog
+#include "logog/include/logog.hpp"
+
+// ** INCLUDES **
+#include "Applications/DataHolderLib/FemCondition.h"
+#include "GeoLib/GEOObjects.h"
+#include "GeoLib/GeoObject.h"
+
+#include "CondItem.h"
+#include "GeoType.h"
+#include "ProcessVarItem.h"
+
+#include <vtkPolyDataAlgorithm.h>
+#include <QFileInfo>
+
+ProcessModel::ProcessModel(DataHolderLib::Project& project, QObject* parent)
+    : TreeModel(parent), _project(project)
+{
+    QList<QVariant> rootData;
+    delete _rootItem;
+    rootData << "Name"
+             << "Value"
+             << ""
+             << ""
+             << "";
+    _rootItem = new TreeItem(rootData, nullptr);
+}
+
+ProcessModel::~ProcessModel() {}
+
+int ProcessModel::columnCount(QModelIndex const& parent) const
+{
+    Q_UNUSED(parent)
+
+    return 2;
+}
+
+void ProcessModel::addConditionItem(DataHolderLib::FemCondition* cond,
+                                    ProcessVarItem* parent)
+{
+    QList<QVariant> item_data;
+
+    item_data << QString::fromStdString(cond->getParamName())
+              << QString::fromStdString(cond->getConditionClassStr());
+
+    CondItem* cond_item = new CondItem(item_data, parent, cond);
+    parent->appendChild(cond_item);
+}
+
+void ProcessModel::addCondition(DataHolderLib::FemCondition* condition)
+{
+    QString name(QString::fromStdString(condition->getProcessVarName()));
+    ProcessVarItem* process_var(getProcessVarItem(name));
+    if (process_var == nullptr)
+        process_var = addProcessVar(name);
+    addConditionItem(condition, process_var);
+}
+
+void ProcessModel::addBoundaryConditions(
+    std::vector<std::unique_ptr<DataHolderLib::BoundaryCondition>> const&
+        conditions)
+{
+    for (size_t i = 0; i < conditions.size(); i++)
+        addCondition(conditions[i].get());
```

Comment:
Please use the for-range-loop when the counting variable is not needed.
```c++
for (auto& c : conditions)
{
    addCondition(*c);
}
```
No need to use `std::unique_ptr<T>::get()`. A `unique_ptr` is a pointer too, so use dereference operator.

prefix `size_t` with the `std` namespace...